### PR TITLE
[GLUTEN-1478][VL] enable timestamp expression tests

### PR DESCRIPTION
**Pull request by rui-mo**
## What changes were proposed in this pull request?
https://github.com/oap-project/gluten/issues/1478

## How was this patch tested?

Testing on CI.

### CODE REVIEW

File: gluten-ut/common/src/test/scala/org/apache/spark/sql/GlutenTestsTrait.scala [hunk 1]
```diff
@@ -227,12 +227,20 @@ trait GlutenTestsTrait extends GlutenTestsCommonTrait {
...
     }
-    if (!(checkResult(result.head.get(0), expected, expression.dataType, expression.nullable)
-      || checkResult(
-      CatalystTypeConverters.convertToCatalyst(result.head.get(0)),
-      CatalystTypeConverters.convertToCatalyst(expected),
-      expression.dataType,
-      expression.nullable))) {
+    val resultToCompare = result.head.get(0) match {
+      case decimal: java.math.BigDecimal =>
+        // Java.math.BigDecimal causes precision < scale issue.
```
Comment (JkSelf) [inline gluten-ut/common/src/test/scala/org/apache/spark/sql/GlutenTestsTrait.scala:232]: This fix seems same with [PR#1697](https://github.com/oap-project/gluten/pull/1697). Can you try locally if PR#1697 can fix this problem? If so, we can merge PR#1697 firstly.

Comment (rui-mo) [reply gluten-ut/common/src/test/scala/org/apache/spark/sql/GlutenTestsTrait.scala:232]: Yes, they are for the same issue. Will rebase after PR#1697 being merged. Thanks for pointing this out!

Comment (JkSelf) [reply gluten-ut/common/src/test/scala/org/apache/spark/sql/GlutenTestsTrait.scala:232]: [PR#1735](https://github.com/oap-project/gluten/pull/1735) merged. And you can rebase it.

Comment (rui-mo) [reply gluten-ut/common/src/test/scala/org/apache/spark/sql/GlutenTestsTrait.scala]: Rebased. Could you review again?

**Pull request merged**